libs/features/personalization/personalization.js

```diff
@@ -1003,32 +1005,40 @@ export async function applyPers(manifests, postLCP = false) {
   config.mep.martech = `|${pznVariants.join('--')}|${pznManifests.join('--')}`;
 }
 
-export const combineMepSources = async (persEnabled, promoEnabled, mepParam) => {
+export const combineMepSources = async (persEnabled, promoEnabled, promoUtilsPromise, mepParam) => {
   let persManifests = [];
+  let initialPersManifestsPromises = [];
+  let initialPersManifests = [];
 
   if (persEnabled) {
-    persManifests = persEnabled.toLowerCase()
+    initialPersManifests = persEnabled.toLowerCase()
       .split(/,|(\s+)|(\\n)/g)
       .filter((path) => path?.trim())
       .map((manifestPath) => ({ manifestPath }));
+    initialPersManifestsPromises = initialPersManifests.map(({ manifestPath }) => {
+      const normalizedURL = normalizePath(manifestPath);
+      return fetch(normalizedURL, { mode: 'same-origin' });
+    });
   }
 
   if (promoEnabled) {
-    const { default: getPromoManifests } = await import('./promo-utils.js');
+    const { default: getPromoManifests } = await promoUtilsPromise;
```

Comment:
we could also move code from promo-utils.js directly here, i think it was not a good decision in the first place to have a separate file.
promo-utils.js contents seem small enough